### PR TITLE
Add unix socket

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This is a competing PR to #466 to *_finally_ get this stuff done.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Please review and tell me what to do.

**Comment by matthiasbeyer** [comment]
And yes, I know it doesn't build... I don't know how to integrate it, yet.

File: src/util/socket.c [hunk 1]
```diff
@@ -36,10 +37,102 @@
...
 
 static struct ws_logger_context log_ctx = { .prefix = "[Sockets Utils] " };
 
+static void
+socket_accept_cb(
+    struct ev_loop* loop,
+    struct ev_io* io,
+    int revents
+) {
+    struct ws_socket* s = (struct ws_socket*) io;
```
Comment (neithernut) [inline src/util/socket.c]: Use the data field in io.

File: src/util/socket.c [hunk 2]
```diff
@@ -25,21 +25,86 @@
...
 #include <sys/socket.h>
 #include <sys/types.h>
 #include <sys/un.h>
+#include <unistd.h>
 
 #include "logger/module.h"
 #include "util/socket.h"
+#include "serialize/json/deserializer.h"
+#include "serialize/json/serializer.h"
+#include "connection/processor.h"
```
Comment (neithernut) [inline src/util/socket.c]: Remove these includes.

**Comment by neithernut** [comment]
ACK for squashing.

**Comment by matthiasbeyer** [comment]
Please review carefully. I squashed almost everything and rewrote the history completely, as you can see.

Hopefully there are no rebase-conflicts left...

**Comment by neithernut** [comment]
ACK.

**Pull request merged**